Trimble.Modus.Components/Controls/TMCheckBox/TMCheckBox.cs

```diff
@@ -0,0 +1,188 @@
+             typeof(bool),
+             typeof(TMCheckBox),
+             true,
+             propertyChanged: OnIsDisabledChanged);
+
+        public static readonly BindableProperty IsIndeterminateProperty =
+          BindableProperty.Create(
+              nameof(IsIndeterminate),
+              typeof(bool),
+              typeof(TMCheckBox),
+              false,
+              propertyChanged: OnIsIndeterminateChanged);
+
+        private static void OnIsIndeterminateChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var customCheckboxView = (TMCheckBox)bindable;
+
+            if (customCheckboxView.IsIndeterminate) { 
+                customCheckboxView.IsChecked = false;
+                customCheckboxView._checkbox.Source = ImageSource.FromResource("Trimble.Modus.Components.Images.indeterminate_checkbox.png");
+
+            }
+            else
+            {
+                customCheckboxView.IsChecked = false;
+            }
+
+        }
+
+        public static readonly BindableProperty TextProperty =
+           BindableProperty.Create(
+               nameof(Text),
+               typeof(string),
+               typeof(TMCheckBox),
+               string.Empty,
+               propertyChanged: OnTextChanged);
+
+
+        public static readonly BindableProperty IsCheckedProperty =
+            BindableProperty.Create(
+                nameof(IsChecked),
+                typeof(bool),
+                typeof(TMCheckBox),
+                false,
+                propertyChanged: OnIsCheckedChanged);
+        public static readonly BindableProperty SizeProperty =
+                   BindableProperty.Create(
+                       nameof(Size),
+                       typeof(CheckboxSize),
+                       typeof(TMCheckBox),
+                       CheckboxSize.Default,
+                       propertyChanged: OnSizeChanged);
+
+        private static void OnSizeChanged(BindableObject bindable, object oldValue, object newValue)
+        {
+            var customCheckboxView = (TMCheckBox)bindable;
+
+            if (customCheckboxView.Size == CheckboxSize.Large)
+            {
+                customCheckboxView._label.FontSize = customCheckboxView._large_font_size; 
+                customCheckboxView._checkbox.WidthRequest = customCheckboxView._large_width; 
+                customCheckboxView._checkbox.HeightRequest = customCheckboxView._large_height; 
+            }
+            else
+            {
+                customCheckboxView._label.FontSize = customCheckboxView._default_font_size;
+                customCheckboxView._checkbox.WidthRequest = customCheckboxView._default_width;
+                customCheckboxView._checkbox.HeightRequest = customCheckboxView._default_height;
+            }
+        }
+
+        public TMCheckBox()
+        {
+            _label = new Label() { FontSize = _default_font_size };
+            _label.VerticalOptions = LayoutOptions.Center;
```

Comment:
Add this vertical option inside label style

Reply:
Formatted